### ISSUE TITLE
Duplexer won't accept school solution

### ISSUE THREAD

**Issue by rleber** [nodeschool/discussions#1480]
I'm having trouble getting the Duplexer problem solved. It even says the school solution is incorrect.

Solution (duplexer.js, same as school solution AFAIK):

``` javascript
var spawn = require('child_process').spawn;
var duplexer = require('duplexer2');

module.exports = function (cmd, args) {
  var ps = spawn(cmd, args);
  return duplexer(ps.stdin, ps.stdout);
};
```

Results:

```
>  stream-adventure verify duplexer.js

TAP version 13
# (anonymous)
ok 1 stream-adventure verify YOURFILE.js
#########################################
###   YOUR SOLUTION IS NOT CORRECT!   ###
#########################################


ok 2 solution exports a function
>
```

Using node v4.1.1 on Mac OS X 10.10.5

**Comment by rleber** [comment]
Poking around a bit, I discovered that the version of duplexer2 I had (0.1.0) was different from the one used by some other modules, e.g. trumpet. 

I forced installation of duplexer2 v0.0.2 and the problem went away.

I'm not sure what changed in duplexer2 between v0.0.2 and v0.1.0 that caused this to break, but I'd be interested to know. I suggest either the tutorial should specify which version of duplexer2 to use, or the school solution should be revised so it works with the newer version.

**Comment by rleber** [comment]
By the way, I was also having problem with Duplexer Redux, using the newer version of duplexer2. Now that I've gone back to the older version, those problems have gone away.

**Comment by finitud** [comment]
I've tried downgrading duplexer2 to v0.0.2 but still get the same problem. Output stops after the first couple of chunks. Using node 4.2.1 on OS X.

**Comment by thisdotrob** [comment]
Also having the same issue

```
ok 1 stream-adventure verify YOURFILE.js
ok 2 solution exports a function
not ok 3 should be equal
  ---
    operator: equal
    expected:
      'azkknnm\nakho\nazjd\naddskd\natqqnv\natf\naxyzmshmd\natf\naxyzmshmd\nanw\nanw\nadptdzsg\naxsd\natqqnv'
    actual:
      'azkknnm\nakho'
    at: ConcatStream.<anonymous> (/usr/local/lib/node_modules/stream-adventure/node_modules/concat-stream/index.js:36:43)
  ...

1..3
# tests 3
# pass  2
# fail  1
```

**Comment by deubaka** [comment]
Same issue, running Mac OS X El Capitan, Node v5.1.0.

```
ok 1 stream-adventure verify YOURFILE.js
ok 2 solution exports a function
not ok 3 should be equal
  ---
    operator: equal
    expected:
      'cbmmppo\ncpccjo\nczuf\ncmjq\ncjuf\ncbu\ncbmmppo\ncvh\ncffumf\nczuf\ncpy\nczuf'
    actual:
      'cbmmppo\ncpccjo'
    at: ConcatStream.<anonymous> (/usr/local/lib/node_modules/stream-adventure/node_modules/concat-stream/index.js:36:43)
```

There was a suggestion in issue comment #1517 on forcing the `stream-combiner2` version to `1.0.2`, but to no success.

**Comment by deubaka** [comment]
Got it working now with multiple options. 

The first one is to downgrade `stream-adventure` to `v3.0.0`, everything works should work there. 

The second be overriding `stream-combiner2` to use `v1.0.2` only as follow,
- Removed module references to `stream-combiner2` latest (v1.1.1), that means removing it in your libs both global and local
- Within the `stream-adventure` module directory, re-install `stream-combiner2` at `v1.0.2` via `npm install  --save stream-combiner@1.0.2`
- Re-run verify and should show a solution passed now

**Comment by reverland** [comment]
Yeah, stuck for a long time here..

    npm install --save stream-combiner2@1.0.2

Thank you, @deubaka

**Comment by kalinchernev** [comment]
Hi, during a triage of the issue queue in the stream-adventure repo, I'd like to suggest another test round on this issue as this [PR has been merged](https://github.com/workshopper/stream-adventure/pull/172) a bit after this issue being lastly communicated on. To my understanding so far, the testing framework had been using dependencies which hadn't been working correctly with duplexer2, but should now.

Also, I opened another pull request to add an options argument in the solution, which appears to be still an issue in Windows for example.

Feedback is welcome, hopefully this is actually fine with the current code base.
